datadog-agent-nvml.yaml

```diff
@@ -0,0 +1,102 @@
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - python-3.11
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - datadog-agent
+      - datadog-agent-core-integrations
+      - rsync
+
+vars:
+  dd_conf: /etc/datadog-agent/conf.d
+  dd_home: / # agent being run by root expects /.
+  dd_shared: /usr/share/datadog-agent
+  python_version: "3.11"
+
+pipeline:
+  # This integration wheel comes from the integrations-extras repository
+  - uses: git-checkout
+    with:
+      repository: https://github.com/DataDog/integrations-extras
+      tag: nvml-${{package.version}}
+      expected-commit: d38c5cdb4ab4d07f4432afb25e0ccd70341efb51
+
+  - runs: |
+      # Create and activate a virtual environment.
+      python -m venv .venv
+      . .venv/bin/activate
+
+      # Use embedded Python in virtual environment.
+      sed -i "s|$(pwd)/.venv|${{vars.dd_shared}}|g" .venv/pyvenv.cfg
+      sed -i "s|$(pwd)/.venv|${{vars.dd_shared}}|g" .venv/bin/*
+
+      # Dump the packages.
+      mkdir site-packages
+      rsync -az ${{vars.dd_shared}}/lib/python${{vars.python_version}}/site-packages/ ./site-packages/
+
+      # This is needed to work around the error "ValueError: ZIP does not support timestamps before 1980"
+      SOURCE_DATE_EPOCH=315532800
+
+      # Install integration.
+      pip install ./nvml
+
+      # Cleanup before preparing the package content.
+      find ${{vars.dd_shared}} -name "*.pyc" -delete
+      find ${{vars.dd_shared}} -name "__pycache__" -exec rm -rf {} +
+
+      # Prepare the package content.
+      diff_packages=$(rsync --include='*/' --exclude='*' --ignore-existing -rv --dry-run ${{vars.dd_shared}}/lib/python${{vars.python_version}}/site-packages/ ./site-packages/ | grep -v "sending incremental file list" | grep -v -E "^\./$|^sent .+ bytes|total size|^$")
+
+      # Include the new packages.
+      packages_dir="${{targets.contextdir}}${{vars.dd_shared}}/lib/python${{vars.python_version}}/site-packages/"
+      mkdir -p "${packages_dir}/datadog_checks"
+      for package in $diff_packages; do
+        rsync -avz "${{vars.dd_shared}}/lib/python${{vars.python_version}}/site-packages/${package}" "${packages_dir}/${package}";
+      done
+
+      # Include the new configurations.
+      conf_dir="${{targets.destdir}}${{vars.dd_conf}}/nvml.d"
+      mkdir -p $conf_dir
+      rsync -avz ${{vars.dd_shared}}/lib/python${{vars.python_version}}/site-packages/datadog_checks/nvml/data/ $conf_dir/
+
+      deactivate
+
+test:
+  environment:
+    contents:
+      packages:
+        - datadog-agent
+        - datadog-agent-core-integrations
+        - datadog-agent-nvml=${{package.full-version}}
+    environment:
+      PYTHONPATH: /usr/share/datadog-agent/lib/python${{vars.python_version}}/site-packages
+      PATH: /opt/datadog-agent/bin/agent/:/opt/datadog-agent/embedded/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/jvm/java-11-openjdk/bi
```

Comment:
We can remove OpenJDK from the path, I believe it's only needed when using JMX

Reply:
Good catch, thanks @EyeCantCU